Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/InstanceUpdateExecutorInterface.h

```diff
@@ -22,13 +22,16 @@ namespace AzToolsFramework
             AZ_RTTI(InstanceUpdateExecutorInterface, "{26BB4ACF-6EA5-4668-84E7-AB75B90BA449}");
             virtual ~InstanceUpdateExecutorInterface() = default;
 
-            // Add all Instances of Template with given Id into a queue for updating them later.
+            // Add an instance into queue for updating later on next system tick.
+            virtual void AddInstanceToQueue(InstanceOptionalReference instance) = 0;
+
+            // Add all instances of template with given id into queue for updating later on next system tick.
             virtual void AddTemplateInstancesToQueue(TemplateId instanceTemplateId, InstanceOptionalConstReference instanceToExclude = AZStd::nullopt) = 0;
```

Comment:
nit: I'd remove the "on next system tick" part from comments, since that's not a guarantee in the long term (when we will start diluting propagations over multiple frames) or in unit tests (where we need to trigger propagation manually)

Reply:
> I'd probably do a pass to see if interfaces can be cached instead of being retrieved via Get on individual functions. Overall it looks good though!

Yup. I am trying out several options. One would be keeping cached focus handler interface in generator (dependency C I mentioned below) and delaying B setup in `PrefabFocusHandler::InitializeEditorInterfaces()` which will be called by `PrefabIntegrationManager::PrefabIntegrationManager()`.